scripts/git_branch.py

```
@@ -16,31 +16,53 @@ def warn(msg):
+            f'{e.stderr.strip()}; {label} suffix will be "unknown"'
+        )
+        return 'unknown'
+    except OSError as e:
+        warn(
+            f'OS error reading git {label}: {e}; '
+            f'{label} suffix will be "unknown"'
+        )
         return 'unknown'
-    except Exception as e:
-        warn(f'Unexpected error reading git branch: {e}; branch suffix will be "unknown"')
+    except Exception as e:  # pylint: disable=broad-exception-caught
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)